### ISSUE TITLE
Issue with ncap2 and using `--output` in versions >4.7.9

### ISSUE THREAD

**Issue by kwilcox** [nco/nco#142]
From looking at the release notes it seems like allowing `ncap2`  to have no input files in `4.8.0` caused a regression in single-file inputs to `ncap2` that was fixed in `4.8.1`. I'm still hitting an issue in `4.8.1`. Note that I'm using the same file for input and output and using the `--output` parameter instead of the positional parameter for the output file. If I use the positional `[out]` parameter it works fine.

## 4.8.1
```
$ ncap2 --version
NCO netCDF Operators version 4.8.1 "INDICES" built by nobody on 2efa0ff75f6b at Jul 12 2019 14:46:52
ncap2 version 4.8.1

$ ncap2 -O -s "defdim(\"dummy\", 1)" --output=file.nc file.nc
ncap2: ERROR file �%i+V not found. It does not exist on the local filesystem, nor does it match remote filename patterns (e.g., http://foo or foo.bar.edu:file).
ncap2: HINT file-not-found errors usually arise from filename typos, incorrect paths, missing files, or capricious gods. Please verify spelling and location of requested file. If the file resides on a High Performance Storage System (HPSS) accessible via the 'hsi' command, then add the --hpss option and re-try command.

$ ncap2 -O -s "defdim(\"dummy\", 1)" file.nc file.nc
(works fine)
```

## 4.7.9
```
$ ncap2 --version
NCO netCDF Operators version 4.7.9 built by nobody on 3e6dc6488ece at Apr 13 2019 16:24:57
ncap2 version 4.7.9

$ ncap2 -O -s "defdim(\"dummy\", 1)" --output=file.nc file.nc
(works fine)

$ ncap2 -O -s "defdim(\"dummy\", 1)" file.nc file.nc
(works fine)
```

**Comment by czender** [comment]
Thanks for reporting this. Will get to it after vacation if @hmb1 does not fix it first.

**Comment by hmb1** [comment]
fixed and  in  master branch

**Comment by czender** [comment]
Tagged and released in 4.8.2-alpha02. @kwilcox please re-open issue if this does not fix behavior for you.
